src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildANetCoreApp.cs

```diff
@@ -1028,5 +1028,48 @@ public void It_warns_on_nonportable_rids(string targetFramework, string[] rids,
                 result.Should().NotHaveStdOutContaining("NETSDK1206");
             }
         }
+
+        [Theory]
+        [InlineData(true, "TRACE DISABLED")]
+        [InlineData(false, "TRACE ENABLED")]
+        public void It_can_use_implicitly_defined_compilation_constants(bool disableTracing, string expectedOutput)
+        {
+            var testProj = new TestProject()
+            {
+                Name = "DisableTracing_" + disableTracing.ToString(),
+                TargetFrameworks = ToolsetInfo.CurrentTargetFramework,
+                IsExe = true,
+            };
+            if (disableTracing == true)
+            {
+                testProj.AdditionalProperties["DisableDiagnosticTracing"] = "true";
+            }
+
+            testProj.SourceFiles[$"{testProj.Name}.cs"] = @"
+using System;
+class Program
+{
+    static void Main(string[] args)
+    {
+        #if TRACE
+            Console.WriteLine(""TRACE ENABLED"");
+        #endif
+        #if !TRACE
+            Console.WriteLine(""TRACE DISABLED"");
+        #endif
+    }
+}";
+            var testAsset = _testAssetsManager.CreateTestProject(testProj);
+
+            var buildCommand = new BuildCommand(Log, Path.Combine(testAsset.Path, testProj.Name));
+            buildCommand
+                .Execute()
+                .Should()
+                .Pass();
+
+            var runCommand = new RunExeCommand(Log, Path.Combine(buildCommand.GetOutputDirectory(ToolsetInfo.CurrentTargetFramework).FullName, $"{testProj.Name}.exe"));
+            var stdOut = runCommand.Execute().StdOut;
+            stdOut.Should().BeEquivalentTo(expectedOutput);
```

Comment:
We have assertions specifically for checking the output of a command:

```suggestion
            runCommand
                .Execute().
                .Should().HaveStdOut(expectedOutput);
```